packages/enricher/src/enricher.ts

```diff
@@ -0,0 +1,101 @@
+import {NanopubWriter} from './writer';
+import {DataFactory} from 'rdf-data-factory';
+import {RdfStore} from 'rdf-stores';
+import {z} from 'zod';
+
+const DF = new DataFactory();
+
+const constructorOptionsSchema = z.object({
+  nanopubWriter: z.instanceof(NanopubWriter),
+});
+
+export type EnricherConstructorOptions = z.infer<
+  typeof constructorOptionsSchema
+>;
+
+const addTextOptionsSchema = z.object({
+  description: z.string(),
+  citation: z.string(), // Required or optional property?
+  about: z.string().url(),
+  creator: z.string().url(),
```

Comment:
Good question! We will probably have to add a new field to the 'Communities' UI. Users can then provide their IRI. This IRI comes from some world-wide user identification system such as [ORCID](https://nl.wikipedia.org/wiki/ORCID). It could also be that only community administrators need to provide this IRI; the IRI will then identify the community, not an individual user.